compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1363,6 +1363,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         "the following trait bounds were not satisfied:\n{bound_list}"
                     ));
                 }
+                if !notes.is_empty() {
+                    notes.iter().for_each(|note| {
+                        err.span_note(span, note.to_string());
+                    });
+                }
```

Comment:
```suggestion
                for note in notes {
                    err.note(note);
                }
```

First, this was written a bit obtusely -- we can just use a for loop 😄. Second, you should use `.note` to be consistent with how trait errors are presented.